tests/protocols/test_calibrationmodel.py

```diff
@@ -86,22 +101,48 @@ def setup_module():
 
 def test_laser_zeroing(setup_module):
     assert setup_module.calmodel is not None
-    fig, ax = plt.subplots(1, 1, figsize=(12, 2))
+    si_peak = 520.45
+    fig, (ax, axpeak, axsifit) = plt.subplots(3, 1, figsize=(12, 4))
     spe_sil_calib = setup_module.calmodel.apply_calibration_x(
         setup_module.spe_sil, spe_units="cm-1"
     )
     setup_module.spe_sil.plot(label="Si original", ax=ax)
     spe_sil_calib.plot(ax=ax, label="Si laser zeroed", fmt=":")
-    # ax.set_xlim(520.45-50,520.45+50)
-
+    plt.grid()
+    ax.set_xlim(si_peak-50, si_peak+50)
     _units = setup_module.calmodel.components[1].model_units
     if _units == "cm-1":
         _units = rf"$\mathrm{{[cm^{-1}]}}$"
     assert _units == "nm"
     ax.set_xlabel(_units)
-    # print(setup_module.calmodel.components[1])
+    spe_test = spe_sil_calib  # .trim_axes(method='x-axis',boundaries=(si_peak-50,si_peak+50))
+    find_kw = {"wlen": 200, "width": 1, "sharpening": None}
+    find_kw["prominence"] = spe_test.y_noise_MAD() * setup_module.calmodel.prominence_coeff
+    cand = spe_test.find_peak_multipeak(**find_kw)
+    fit_kw = {}
+    fitres = spe_test.fit_peak_multimodel(profile=setup_module.si_profile, candidates=cand, **fit_kw, no_fit=False)
+    fitres.plot(ax=axpeak, label="fit res")
+    spe_test.plot(ax=axpeak, label="Si laser zeroed", fmt=":")
+
+    axpeak.set_xlim(si_peak-50, si_peak+50)
+
+    # let's look at the peak before laser zeroing
+    spe_test_necalibrated_only = setup_module.calmodel.components[0].process(spe_test)
+    setup_module.calmodel.components[1].fit_res.plot(ax=axsifit)
+    si_peak_nm = setup_module.calmodel.components[1].model
+    axsifit.set_xlim(si_peak_nm-5, si_peak_nm+5)
+    spe_test_necalibrated_only.plot(ax=axsifit, label="Si (Ne calibrated only)", fmt=":")
+
+    df = fitres.to_dataframe_peaks().sort_values(by="height", ascending=False)
+    print(df.iloc[0]["center"])
+
+    print(df.sort_values(by="amplitude", ascending=False).head())
+
+    plt.grid()
     plt.tight_layout()
     plt.savefig("test_calmodel_{}.png".format("laser_zeroing"))
```

Comment:
Do you really need to generate plots in the tests? They are slow and do not believe that they are going to be used somehow as part of the tests.